auth_user_insensitive/models/res_users.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015-2017 LasLabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class ResUsers(models.Model):
+
+    _inherit = 'res.users'
+
+    login = fields.Char(
+        'Login',
+        size=64,
+        required=True,
+        help='Used to log into the system. Case insensitive.',
+    )
+
+    @api.model
+    def search(self, domain, *args, **kwargs):
+        """ Overload search to lowercase name domains. This can't be done in
+        a regular search method as the field is not computed
```

Comment:
But isn't ILIKE the default operator used both on the search view and on the authentication method? If so, that's all we need.